firebase-firestore/src/main/java/com/google/firebase/firestore/LoadBundleTaskProgress.java

```diff
@@ -27,39 +38,79 @@
   private int totalDocuments;
   private long bytesLoaded;
   private long totalBytes;
-  private TaskState taskState;
+  @NonNull private final TaskState taskState;
+  @Nullable private final Exception exception;
 
   public LoadBundleTaskProgress(
       int documentsLoaded,
       int totalDocuments,
       long bytesLoaded,
       long totalBytes,
+      @Nullable Exception exception,
       @NonNull TaskState taskState) {
     this.documentsLoaded = documentsLoaded;
     this.totalDocuments = totalDocuments;
     this.bytesLoaded = bytesLoaded;
     this.totalBytes = totalBytes;
     this.taskState = taskState;
+    this.exception = exception;
   }
 
+  /** Returns how many documents have been loaded. */
   public int getDocumentsLoaded() {
     return documentsLoaded;
   }
 
+  /**
+   * Returns the total number of documents in the bundle. Returns 0 if the bundle failed to parse.
+   */
   public int getTotalDocuments() {
     return totalDocuments;
   }
 
+  /** Returns how many bytes have been loaded. */
   public long getBytesLoaded() {
     return bytesLoaded;
   }
 
+  /** Returns the total number of bytes in the bundle. Returns 0 if the bundle failed to parse. */
   public long getTotalBytes() {
     return totalBytes;
   }
 
+  /** Returns the current state of the task. */
   @NonNull
   public TaskState getTaskState() {
     return taskState;
   }
+
+  /** If the task failed, returns the exception. Otherwise, returns null. */
+  @Nullable
+  public Exception getError() {
+    return exception;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    LoadBundleTaskProgress that = (LoadBundleTaskProgress) o;
+
+    if (documentsLoaded != that.documentsLoaded) return false;
+    if (totalDocuments != that.totalDocuments) return false;
+    if (bytesLoaded != that.bytesLoaded) return false;
+    if (totalBytes != that.totalBytes) return false;
+    return taskState == that.taskState;
+  }
+
+  @Override
+  public int hashCode() {
+    int result = documentsLoaded;
+    result = 31 * result + totalDocuments;
```

Comment:
Consider using `Objects.hash()` from the standard library instead of implementing your own hash code.

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...-

Reply:
It's not available with Android 14 (requires API level 19+).